pkg/image/oci/directory_provider.go

```diff
@@ -31,20 +31,29 @@ func (p *DirectoryImageProvider) Provide() (*image.Image, error) {
 		return nil, fmt.Errorf("unable to parse OCI directory index: %w", err)
 	}
 
-	manifest, err := index.IndexManifest()
+	indexManifest, err := index.IndexManifest()
 	if err != nil {
-		return nil, fmt.Errorf("unable to parse OCI directory manifest: %w", err)
+		return nil, fmt.Errorf("unable to parse OCI directory indexManifest: %w", err)
 	}
 
-	// for now, lets only support one image manifest (it is not clear how to handle multiple manifests)
-	if len(manifest.Manifests) != 1 {
-		return nil, fmt.Errorf("unexpected number of OCI directory manifests (found %d)", len(manifest.Manifests))
+	// for now, lets only support one image indexManifest (it is not clear how to handle multiple manifests)
+	if len(indexManifest.Manifests) != 1 {
+		return nil, fmt.Errorf("unexpected number of OCI directory manifests (found %d)", len(indexManifest.Manifests))
 	}
 
-	img, err := pathObj.Image(manifest.Manifests[0].Digest)
+	manifest := indexManifest.Manifests[0]
+	img, err := pathObj.Image(manifest.Digest)
 	if err != nil {
 		return nil, fmt.Errorf("unable to parse OCI directory as an image: %w", err)
 	}
 
-	return image.NewImage(img), nil
+	var metadata []image.AdditionalMetadata
+
+	// make a best-effort attempt at getting the raw indexManifest
+	rawManifest, err := img.RawManifest()
+	if err == nil {
+		metadata = append(metadata, image.WithManifest(rawManifest))
+	}
+
+	return image.NewImage(img, metadata...)
```

Comment:
we may have 0 to 1 optional metadata provided to the image, `metadata` is always used, it just may be empty sometimes

Reply:
Okay, yeah. I do see how one could consider the presence or absence of an optional metadata function to be thought of as a collection that can contain zero items or one item.

But I think there's a more obvious interpretation of what's happening. We can create a new image with or without a manifest. This view leads to a more concise, and IMHO, a more idiomatic implementation — where there's no need to unpack an empty slice into zero parameters because it was declared but sometimes it never ends up getting populated, and when it _is_ populated, it's only ever with a single value.

For example,

```go
rawManifest, err := img.RawManifest()
if err == nil {
  return image.NewImage(img, image.WithManifest(rawManifest))
}

return image.NewImage(img)
```

instead of

```go
var metadata []image.AdditionalMetadata

rawManifest, err := img.RawManifest()
if err == nil {
  metadata = append(metadata, image.WithManifest(rawManifest))
}

return image.NewImage(img, metadata...)
```

Reply:
Side note: I missed the opportunity to provide the manifest digest as well here, just added this.

This is a style choice, yes? Also I consider the second to be more defensive --the image is created in one spot, not two, so there is only one place where to alter the call arguments.

Reply:
Okay cool, I'll check out the new version. I don't consider this a style choice in the sense that the implementation choice is arbitrary or that the consequences are the same between the choices, but I'll look at the new version before commenting further!

Reply:
Okay gotcha. I think the use of a slice is more motivated now, with there being more than one possible slice item.

It still looks like the approach that foregoes the slice is more concise, and the conditional paths are more apparent. As written, I think it would take a senior developer at least a couple of passes to understand what's really going on here, which is something I'd like us to avoid if possible.

I'll approve this. I appreciate you explaining your thinking, and for pointing out the code change. I'd like us to talk more about various approaches to code (as a team) as we move forward. Happy to chat more about this in Zoom or something if you'd like. :)